### DISCUSSION TITLE
Applying environment exports by <source *.sh> on meson build

### DISCUSSION THREAD

**Discussion by Anton-K-dev** [mesonbuild/meson#11841] [General]
OS: Linux

Can anyone help, how to apply exports in meson.build during setup as I do it on terminal via 
`source environment.sh`?

Also I have environment script from external SDK and don't want to change mode it executable.

Thanks in advance, Anton.

**Comment by eli-schwartz** [comment]
An environment.sh script sounds like the kind of thing you're supposed to use to set up a development shell *before* running `meson setup`.

**Reply by Anton-K-dev** [reply]
> An environment.sh script sounds like the kind of thing you're supposed to use to set up a development shell _before_ running `meson setup`.

That's correct. But can I add it to the meson.build instead of calling that before?

**Reply by eli-schwartz** [reply]
I'm not sure how it could -- meson isn't written in bash so it can't source the file, and forking a bash process to run it wouldn't have any effect on meson itself, only the forked subprocess.

**Comment by eli-schwartz** [comment]
What do you need these environment variables for? Is it possible to convert the same information to a meson native/cross file e.g. by setting paths to compiler binaries?

**Reply by Anton-K-dev** [reply]
> What do you need these environment variables for? Is it possible to convert the same information to a meson native/cross file e.g. by setting paths to compiler binaries?

Ok, thank you anyway. I don't need it much enough
